openssl.go

```diff
@@ -268,35 +269,50 @@ func base(b []byte) *byte {
 	return unsafe.SliceData(b)
 }
 
-func newOpenSSLError(msg string) error {
+// newMkcgoErr creates a new error from the given mkcgo_err_state
+// and frees the state. If errst is nil, it returns nil.
+func newMkcgoErr(msg string, errst *C.mkcgo_err_state) error {
+	if errst == nil {
+		return nil
+	}
+	defer C.mkcgo_err_free(errst)
 	var b strings.Builder
 	b.WriteString(msg)
 	b.WriteString("\nopenssl error(s):")
-	for {
-		var (
-			e    uint32
-			file *byte
-			line int32
-		)
-		switch vMajor {
-		case 1:
-			e = go_openssl_ERR_get_error_line(&file, &line)
-		case 3:
-			e = go_openssl_ERR_get_error_all(&file, &line, nil, nil, nil)
-		default:
-			panic(errUnsupportedVersion())
-		}
+	for i := range 16 {
+		e := uint32(errst.code[i])
 		if e == 0 {
 			break
 		}
 		b.WriteByte('\n')
 		var buf [256]byte
 		go_openssl_ERR_error_string_n(e, base(buf[:]), len(buf))
-		b.WriteString(string(buf[:]) + "\n\t" + C.GoString((*C.char)(unsafe.Pointer(file))) + ":" + strconv.Itoa(int(line)))
+		b.Write(buf[:])
```

Comment:
Maybe fix null terminator handling in buf while here?

```suggestion
		if termIdx := bytes.IndexByte(buf[:], 0); termIdx != -1 {
			b.Write(buf[:termIdx])
		} else {
			b.Write(buf[:])
		}
```

(Brief mention in https://github.com/golang-fips/openssl/issues/266.)